lib/util/util.c

```diff
@@ -895,6 +895,13 @@ char * util_alloc_realpath__(const char * input_path) {
         if (path_index == path_len)
           break;
 
+        /* We have backrefed us out of the current filesystem. */
+        if (prev_index < 0) {
+          free( abs_path );
+          free( real_path);
+          return NULL;
```

Comment:
Is this _really_ desired behaviour? That it returns `nullptr` when it hits "beyond root"?

Reply:
I agree that I would expect `../../../../../../../..adinfinitum` to return root since `/.. = /`.

Reply:
OK - I just checked this:
```
cd /tmp/path
cd ../../../../../../tmp/path
pwd 
/tmp/path
```
I must say I was a little surprised by that - but then yes I fully agree that the semantics should be changed; thank you.